File: Joanna_Laurence/src/main/java/org/uva/sea/languages/qls/parser/elements/specification/Section.java
package org.uva.sea.languages.qls.parser.elements.specification;

import org.antlr.v4.runtime.Token;
import org.uva.sea.languages.qls.parser.visitor.IStyleASTVisitor;

import java.util.List;

public class Section extends Specification {

    private final String name;
    private final List<Specification> specifications;

    public Section(final Token token, final String name, final List<Specification> specifications) {
        super(token);
        this.name = name;
        this.specifications = specifications;
    }

    public final String getName() {
        return this.name;
    }

    public final Iterable<Specification> getSpecifications() {
        return this.specifications;
    }

    @Override
    public final <T> T accept(final IStyleASTVisitor<T> visitor) {
        return visitor.visit(this);
    }
}

File: DogeQL/src/main/kotlin/qls/ast/node/Section.kt
package qls.ast.node

import qls.common.Name
import qls.visitor.QlsVisitor

data class Section(val title: Name, val elements: List<Element>) : Element, Style {
    override fun <T> accept(visitor: QlsVisitor<T>): T {
        return visitor.visit(this)
    }
}